pkgs/tools/virtualization/nixos-shell/default.nix

```
@@ -0,0 +1,30 @@
+{ stdenv, nix, fetchFromGitHub, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  pname = "nixos-shell";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "Mic92";
+    repo = "nixos-shell";
+    rev = version;
+    sha256 = "1qk5a01vh6wbbkib8xr57w1j4l3n6xdjd46nsw9bsa444fzlc0wr";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postInstall = ''
+    wrapProgram $out/bin/nixos-shell \
+      --prefix PATH : ${stdenv.lib.makeBinPath [ nix ]}
```

Isn't it safe to assume a user of this program will have `nix` executables in his `$PATH`? (I'm ready to hear statements such as I got in #88136 :)).

---

I'm just wondering if it's worth making this PR a draft because of infinite recursion which I suspect this line is the cause of it.

Perhaps this will work:

```suggestion
      --prefix PATH : ${nix}/bin
```

---

No the infinite recursion error is actually in NixOS itself. I only get it with a specific NIX_PATH. I don't think it is specific to this program since I am also getting it in other evaluations. I tracked it down to `revisionWithDefault` in `lib/trivial.nix`.

---

It is save to assume that we have nix in PATH but it is not safe to assume that it can evaluate the nix code that is provided by the program. Recompilation of this program is fast in case anyone wants to rebuild it.